code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForApiAuthorizationServlet.java

```diff
@@ -0,0 +1,128 @@
+import java.util.logging.Logger;
+
+/**
+*  This class directs the creation of user Credentials for accessing APIs.
+*/
+
+@WebServlet("/check-for-api-authorization")
+public abstract class CheckForApiAuthorizationServlet extends HttpServlet {
+
+  abstract void useCredential(Credential credential);
+  // should return "check-for-api-authorization" for now
+  abstract String getServletURIName();
+  //Update the userId with the newly created credential
+  abstract void updateUser(String userId);
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final List<String> SCOPES = Collections.singletonList("https://www.googleapis.com/auth/contacts.readonly");
+  private static final String CREDENTIALS_FILE_PATH = "WEB-INF/credentials.json";
+  private static final String url = "https://8080-ac896ae1-5f0c-45b5-8dfe-19fa4d3d8699.us-east1.cloudshell.dev";
+  static final Logger log = Logger.getLogger(CheckForApiAuthorizationServlet.class.getName());
+
+  //Creates the user's credential
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get flow for token response
+    GoogleAuthorizationCodeFlow flow = getFlow(response);
+    if (flow != null) {
+      String code = request.getParameter("code");
+      String idTokenString = request.getParameter("state");
+
+      // Get userId
+      String userId = getUserId(idTokenString, flow, response);
+      if (userId != null) {
+        TokenResponse tokenResponse = flow.newTokenRequest(code).setRedirectUri(url+getServletURIName()).execute();
+        Credential credential = flow.createAndStoreCredential(tokenResponse, userId);
+        updateUser(userId);
+        useCredential(credential);
+      }
+    }
+  }
+  
+  //Creates the url for authorizing the user
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get token to pass into redirect
+    String idToken = request.getParameter("idToken");
+    // Get flow to build url redirect
+    GoogleAuthorizationCodeFlow flow = getFlow();
+
+    if (flow != null) {
+      AuthorizationRequestUrl authUrlRequestProperties = flow.newAuthorizationUrl().setScopes(SCOPES).setRedirectUri(url+getServletURIName()).setState(idToken);
+      String url = authUrlRequestProperties.build();
+      // Send url back to client
+      response.getWriter().println(url);
+    }
+  }
+
+  public String getUserId(String idTokenString, GoogleAuthorizationCodeFlow flow, HttpServletResponse response) {
+    try {
+      NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+      
+      // Make verifier to get payload
+      GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(HTTP_TRANSPORT, JSON_FACTORY)
+        .setAudience(Collections.singletonList("1080865471187-u1vse3ccv9te949244t9rngma01r226m.apps.googleusercontent.com"))
```

Comment:
Put this insane string into a static final variable most likely if it is supposed to be a hardcoded value. Give it a good variable name.